dccpi_mm/dcc_loco_control.py

```diff
@@ -0,0 +1,98 @@
+import redis
+import time
+import json
+
+from .dcc_logger         import getLogger
+from .dcc_packet_factory import DCCPacketFactory
+from .dcc_hardware       import DCCHardware
+from .dcc_redis_queue    import RedisQueue
+
+class DCCKeyboardLocoControl(object):
+    """
+    Class id defined to be used in console application
+    and and handle keys
+    By default arrow keys up/down areused for speed control,
+    and numbers are used to control functions
+    """
+    def __init__(self, loco_address, commands_queue_name, 
+                 emergency_queue_name, **redis_kwargs):
+
+        self.lolco_address = loco_address
+        self.commands_queue = RedisQueue(commands_queue_name, **redis_kwargs)
+        self.emergency_queue = RedisQueue(emergency_queue_name, **redis_kwargs)
+        self.loco_functions = {
+            'FL' : 0,
+            "Fn1": 0,
+            "Fn2": 0,
+            "Fn3": 0,
+            "Fn4": 0,
+        }
+        self.loco_speed = 0
+        self.loco_address = loco_address
+        self.loco_direction = 'undef'
+        # Prededefined keys
+        self.UP_SPEED_KEYS = ('up')
+        self.DOWN_SPEED_KEYS = ('down')
+
+        self.logger = getLogger("DCCLococControl")
+        self.logger.debug('Init')
+
+    def exit(self):
+        """
+        Base control station send eStop packet on track if any data in
+        emergency queue
+        so any payload can be used for emergency packet
+        """
+        self.emergency_queue.put('{ "action": "emergency_stop" }')
+
+    def key_handler(self, key):
+        """
+        Simple key handler
+        """
+        if key in self.UP_SPEED_KEYS:
+            # Set direction if not defined.
+            # Direction can be changed only if speed = 0
+            if ( self.loco_speed == 0 ) and ( self.loco_direction != 'forward' ):
+                self.loco_direction = 'forward'
```

Comment:
> I would not keep "direction" marker, if it depends only from "speed" value (above or belove 0) (you can make calculated property (@Property) for easy access to this value).

It can't be calculated from speed.
"Forward Lamp" function will switch on/off lamp depends on direction even if speed is Zero.